Update SQL limitations page for release 2.8 features (#4265)
opensearch-project/documentation-website: _search-plugins/sql/limitation.md

### V2 engine limitations

* The [cursor feature](#pagination-only-supports-basic-queries) is supported by the `V1` engine only.
  * For support of `cursor`/`pagination` in the `V2` engine, track [GitHub issue #656](https://github.com/opensearch-project/sql/issues/656).
* `json` formatted output is supported in `V1` engine only. 
* The `V2` engine does not track query execution time, so slow queries are not reported.

> * The `V2` query engine not only runs queries in the OpenSearch engine but also supports post-processing for complicated queries. Accordingly, the `explain` output is no longer pure OpenSearch domain-specific language (DSL) but also includes query plan information from the `V2` query engine.

Reviewer: ```suggestion
* The `V2` query engine not only runs queries in the OpenSearch engine but also supports post-processing for complex queries. Accordingly, the `explain` output is no longer OpenSearch domain-specific language (DSL) but also includes query plan information from the `V2` query engine.
```

> * The `V2` query engine does not support aggregation queries (`histogram`, `date_histogram`, `percentiles`, `topHits`, `stats`, `extended_stats`, `terms`, `range`)

Reviewer: ```suggestion
* The `V2` query engine does not support aggregation queries such as`histogram`, `date_histogram`, `percentiles`, `topHits`, `stats`, `extended_stats`, `terms`, or `range`.
```

> * The `V2` query engine does not support aggregation queries (`histogram`, `date_histogram`, `percentiles`, `topHits`, `stats`, `extended_stats`, `terms`, `range`)

Author: ```suggestion
Suggested change
* The `V2` query engine does not support aggregation queries such as `histogram`, `date_histogram`, `percentiles`, `topHits`, `stats`, `extended_stats`, `terms`, or `range`.
```

* JOINs and sub-queries are not supported, track [GitHub issue #1441](https://github.com/opensearch-project/sql/issues/1441) and [GitHub issue #892](https://github.com/opensearch-project/sql/issues/892).

> * PartiQL syntax for `nested` queries are not supported.  Additionally, arrays of objects and primitive types return the first index of the array, while in `V1` they return the entire array as a json object. 

Reviewer: ```suggestion
* PartiQL syntax for `nested` queries are not supported.  Additionally, arrays of objects and primitive types return the first index of the array, while in `V1` they return the entire array as a JSON object. 
```

> * The [cursor feature](#pagination-only-supports-basic-queries) is supported by the `V1` engine only.

Reviewer 2: Maybe update this line too?

For support of `cursor`/`pagination` in the `V2` engine, track [GitHub issue #656](https://github.com/opensearch-project/sql/issues/656).

> * `json` formatted output is no longer supported. 

Reviewer 2: It is supported in V1 only

* The `V2` engine does not track query execution time, so slow queries are not reported.

> * The `V2` query engine not only runs queries in the OpenSearch engine but also supports post-processing for complicated queries. Accordingly, the explain output is no longer pure OpenSearch domain-specific language (DSL) but also includes query plan information from the `V2` query engine.

Reviewer 2: mention `nested` and meta-fields?

Author: This is for limitations.  I don't think I need to write this as a limitation in V1.

* The `V2` query engine does not support aggregation queries (historgram, date_histogram, percentiles, topHits, stats, extended_stats, terms, range)

> * JOINs are sub-queries are not supported, track [GitHub issue #1441](https://github.com/opensearch-project/sql/issues/1441) and [GitHub issue #892](https://github.com/opensearch-project/sql/issues/892).

Reviewer 3: ```suggestion
* JOINs and sub-queries are not supported, track [GitHub issue #1441](https://github.com/opensearch-project/sql/issues/1441) and [GitHub issue #892](https://github.com/opensearch-project/sql/issues/892).
```
